app/services/action_handlers/pull_request.rb

```diff
@@ -0,0 +1,40 @@
+module ActionHandlers
+  class PullRequest < ActionHandler
+    ACTIONS = %w[open review_requested closed \
+                 review_request_removed].freeze
+    private_constant :ACTIONS
+
+    private
+
+    def handleable?
+      ACTIONS.include?(@payload['action'])
+    end
+
+    def open
+      @event_type.open!
+      @event_type.update!(opened_at: Time.current)
+    end
+
+    def merged
+      @event_type.update!(merged_at: Time.current)
+    end
+
+    def closed
+      merged if @payload['pull_request']['merged'] == true
+      @event_type.closed!
+      @event_type.update!(closed_at: Time.current)
+    end
+
+    def review_request_removed
+      reviewer = find_user
```

Comment:
Why can't we use `find_or_create_user` here?

Reply:
I thought of forcing to find a user, because in that part the user should already exist, but in the other hand, would be a problem if the user does not exist?

Reply:
I guess if there's no user at this point, it could be a symptom of a bug in the code. Following that logic, it's fine to keep the method. I would move it to this class though, since it's only used here.

Reply:
Ok!